src/pygeoprocessing/symbolic.py

```diff
@@ -188,10 +190,18 @@ def _generic_raster_op(*arg_list):
             raise ValueError(
                 "`target_nodata` is undefined (None) but there are nodata "
                 "values present in the input rasters.")
-        func_result = func(*[array[valid_mask] for array in array_list])
+        user_symbols = dict((symbol, array[valid_mask]) for (symbol, array) in
+                            zip(kwarg_names, array_list))
     else:
         # there's no nodata values to mask so operate directly
-        func_result = func(*array_list)
+        user_symbols = dict((symbol, array) for (symbol, array) in
+                            zip(kwarg_names, array_list))

```

Comment:
Won't need this if you like the above.

```suggestion
```

Reply:
Nearly!  The `{ ... }` notation doesn't quite work here because `{zip(...)}` yields a `set` with  `zip` object in it.

Instead, I've updated this to be `dict(zip(...))` and that works great and avoids the unnecessary comprehension.  Updated in 05b0195